### ISSUE TITLE
The active test run was aborted. Reason: After parsing a value an unexpected character was encountered

### ISSUE THREAD

**Issue by davidfowl** [Microsoft/vstest#1542]
This is another version of https://github.com/Microsoft/vstest/issues/1435 that we're seeing recently on various SignalR runs. I don't have a consistent repro but here's an instance of that failure:

https://ci.appveyor.com/project/aspnetci/signalr/build/1.0.4105

```
otnet.exe : The active test run was aborted. Reason: After parsing a value an unexpected character was encountered: V. Path 'Payload.NewTestResults[1].Messages[0].Text', line 1, position 16386.
At C:\Users\appveyor\.dotnet\buildtools\korebuild\2.1.0-preview3-17002\scripts\common.psm1:7 char:5
+     & $_cmd @args
+     ~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (The active test...position 16386.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
Total tests: Unknown. Passed: 1139. Failed: 0. Skipped: 11.
Test Run Aborted.
```

The current commit that reproduces the problem https://github.com/aspnet/SignalR/commit/d8d8e24c80c94dd78aef20cdf9a851aba13d7625, just run build.cmd or build.sh and hope you get lucky. 

PS: These types of errors are super annoying when they happen because they probably are masking actual issues.

**Comment by davidfowl** [comment]
Happened again here https://ci.appveyor.com/project/aspnetci/signalr/build/1.0.4149

**Comment by davidfowl** [comment]
Happened again here https://ci.appveyor.com/project/aspnetci/signalr/build/1.0.4302

**Comment by davidfowl** [comment]
@smadala what can I do to get somebody to investigate this?

**Comment by smadala** [comment]
@jayaranigarg @abhishekkumawat23 Can you please look into this issue?

**Comment by jayaranigarg** [comment]
Looking into this..

**Comment by jayaranigarg** [comment]
@davidfowl : I followed following repro steps
1. Cloned repo and went to commit 4a568e90d2eba112b7adae3526ef6
2. Ran build.cmd
and hit this error:
![image](https://user-images.githubusercontent.com/17309452/39054435-3f83373a-44cf-11e8-8b78-1bfd79323fb1.png)
What did I miss out?

**Comment by davidfowl** [comment]
You need to install nodejs https://nodejs.org/en/.

**Comment by jayaranigarg** [comment]
@davidfowl : After installing nodejs I am facing these errors:

![image](https://user-images.githubusercontent.com/17309452/39127457-b7155fb2-4722-11e8-883c-efb2c052fb68.png)

[2018-04-23T09_49_03_085Z-debug.log](https://github.com/Microsoft/vstest/files/1938220/2018-04-23T09_49_03_085Z-debug.log)

**Comment by anurse** [comment]
That error is in our browser-based JavaScript tests, which run after the `dotnet test`-based test, so the error would have reproed before getting to that point. It looks like the screenshot is cut off below the actual error so I'm not able to see why the browser tests failed, but for this investigation you can disregard failures at that point.

Note that the issue with `dotnet test` does not generally repro on our dev machines, only on the (generally much slower) machines on the [AppVeyor CI](http://appveyor.com/) platform we use to test pull requests.

Is there any way to enable more diagnostics to see what is failing to parse? Since the issue is so difficult to repro on a dev machine, having more diagnostics might make it easier for us to get your more details on what is causing the error in the environment we are using.

**Comment by cltshivash** [comment]
@BrennanConroy @anurse  Please enable diag logs https://github.com/Microsoft/vstest-docs/blob/master/docs/diagnose.md#dotnet-test

**Comment by BrennanConroy** [comment]
Reproduced the issue and got diag logs
https://microsoft-my.sharepoint.com/:u:/p/brecon/EUXU80VCVJBEhw2kjnkhD74BC5zjm4PeVWEF-ZqG0PPhNw?e=RveBhQ

**Comment by cltshivash** [comment]
@BrennanConroy Thanks for the logs.

**Comment by mayankbansal018** [comment]
@BrennanConroy , I was analyzing these logs, & found that while reading from network stream, we are reading data from multiple writes. To verify this, we would need to get logs from the point where we are writing data into network stream, which would confirm that whether a corrupted data was written, or whether the issue is during read.
The change to log write data, is present in our latest 15.7 version, but unfortunately dotnet sdk has not released from the branch where we inserted our platform.
Alternate way would be that we provide you some patched dll's which would log this additional information for us, let me know if patching with our dll's in your CI/CD would be possible?

**Comment by mayankbansal018** [comment]
@davidfowl , @BrennanConroy , can you please share from which branch of dotnet/cli do you pick up dotnet sdk? From the logs it seems that it is from https://github.com/dotnet/cli/tree/release/2.1.3xx , our test platform logging changes went in https://github.com/dotnet/cli/tree/release/2.1.2xx, but since no release has been made from it, we can't get additional logs from your CI/CD.

I'll push latest TP in the dotnet branch from where you pick dotnet sdk.

**Comment by anurse** [comment]
We use builds from the 2.1.300 build of the SDK, which is the most recent build as far as I'm aware. @natemcmaster can clarify exactly where we get our CLI builds from.

**Comment by natemcmaster** [comment]
We pick up CLIs from the 2.1.3xx branch.

> From the logs it seems that it (vstest) is from 2.1.3xx , which is slightly older, & our test platform logging changes went in 2.1.2xx

This is concerning to me. 2.1.3xx has shipped two previews. We'll have an RC and final build very soon. If the vstest bundled in 2.1.3xx is older than what you see in 2.1.2xx, we should make sure to fix it ASAP. @johnbeisner @livarcocc can help you get the right branches merged/versions updated, etc.

**Comment by livarcocc** [comment]
What version should we be inserting for 2.1.3xx?

**Comment by mayankbansal018** [comment]
@livarcocc 2.1.3xx should be on [15.7.0](https://github.com/dotnet/cli/blob/b390c9805e6730886129e424f9ce23ee4604266c/build/DependencyVersions.props#L44) , but it is on [15.7.0-preview-20180221-13](https://github.com/dotnet/cli/blob/release/2.1.3xx/build/DependencyVersions.props#L47)

We made change in 2.1.2xx around a month back, & I was hoping it would flow in 3xx implicitly, should I've raised a PR https://github.com/dotnet/cli/pull/9191, can you please merge it.

**Comment by mayankbansal018** [comment]
I ran test for assembly "Microsoft.AspNetCore.SignalR.Tests", as these were reported failing in logs, on a machine multiple times(20-30 odd) on a low core isolated machine, but the tests did not fail once. I'll continue working on this on Monday.

**Comment by mayankbansal018** [comment]
@davidfowl , @BrennanConroy I analyzed this further, & the issue is happening because you are still on older version of testplatform, which does not have communication fix. We have pushed this fix in dotnet cli release/2.1.2xx branch, but it has not flown to release/2.1.3xx.

@livarcocc I've raised a PR I've raised a PR https://github.com/dotnet/cli/pull/9191 for release/2.1.3xx, can you please approve & complete it.

**Comment by JamesNK** [comment]
@mayankbansal018 3xx already has the change apparently. What does that mean for us getting the fix?

**Comment by mayankbansal018** [comment]
@JamesNK , once there is a release from 3xx, & you get that dotnet sdk picked for running tests, this issue will go away.

**Comment by mayankbansal018** [comment]
Closing this issue, it will get resolved after a dotnet sdk release from 2.1.3xx branch
